server/controllers/userAuth.js

```diff
@@ -126,9 +126,74 @@ const signUp = (req, resp, next) => {
 
 const confirmEmail = (req, resp) => {};
 
+const resetPassword = (req, resp, next) => {
+  const resetToken = crypto.randomBytes(32).toString('hex');
+  const resetTokenExpirationDate = new Date().getTime() + 3600000;
+  const { email } = req.body;
+  const sanitizedEmail = sanitize(email);
+  const { isEmail, isEmpty } = validator;
+
+  if (isEmpty(sanitizedEmail) || !isEmail(sanitizedEmail)) {
+    return resp
+      .status(400)
+      .send({ message: 'authorization.actions.wrong-email' });
+  }
+
+  User.findOne({ email: sanitizedEmail })
+    .exec()
+    .then(user => {
+      if (!user) {
+        throw new Error('authorization.actions.no-user-of-email');
```

Comment:
proposal: what do you think about some route params, redirect to them and then using this param on frontend (match: { params: { your-param }}}) and decide about messageId:
frontend:
```
<Route component={ResetPassword} path="/reset-password/:result?" />
```
backend:
```
return resp.redirect('/reset-password/no-user');
return resp.redirect('/reset-password/provider-user');
return resp.redirect('/reset-password/inactive-user');
```

Reply:
I haven't seen such a solution for handling errors.
I don't think it is a proper way to do it. 

@marekrozmus what do you think?